### DISCUSSION TITLE
[Bug]: V1.get_conversations() response 404

### DISCUSSION THREAD

**Discussion by cjb1234567** [acheong08/ChatGPT#1112] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

V1
chat_bot.ask() returns revChatGPT.V1.Error
because url returns 404

### Steps to reproduce the problem

1. Go to ....
V1.get_conversations()

### What should have happened?

 url = BASE_URL + f"api/conversations?offset={offset}&limit={limit}"
>> 'https://apps.openai.com/api/conversations?offset=0&limit=20'
returns 404

### Version where the problem happens

V1
Version: 2.3.5

### What Python version are you running this with?

3.10

### What is your operating system ?

Linux

### Command Line Arguments

```Shell
no
```

### Console logs

```Shell
File "E:\DockerProj\gptweb\venv\Lib\site-packages\revChatGPT\V1.py", line 144, in ask
    self.__map_conversations()
  File "E:\DockerProj\gptweb\venv\Lib\site-packages\revChatGPT\V1.py", line 296, in __map_conversations
    conversations = self.get_conversations()
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\DockerProj\gptweb\venv\Lib\site-packages\revChatGPT\V1.py", line 236, in get_conversations
    self.__check_response(response)
  File "E:\DockerProj\gptweb\venv\Lib\site-packages\revChatGPT\V1.py", line 226, in __check_response
    raise error
revChatGPT.V1.Error
```

**Comment by acheong08** [comment]
Recommendation: Use V3.

**Comment by acheong08** [comment]
It is due to the endpoint being shut down by OpenAI

**Comment by fish8848-chen** [comment]
> It is due to the endpoint being shut down by OpenAI


Is this the endpoint you're talking about?  https://chat.duti.tech/api/

**Comment by Pengchengistaken** [comment]
> > It is due to the endpoint being shut down by OpenAI
> 
> Is this the endpoint you're talking about? https://chat.duti.tech/api/

most this kind of endpoint

**Comment by 7trail** [comment]
I'm trying to find free alternatives. I don't want to pay for ChatGPT [obviously], so I'm looking into my options.

**Comment by acheong08** [comment]
Working on it. Could exist other bypasses.

**Comment by 7trail** [comment]
I'm thinking the same thing @acheong08 . Lmk if you need assistance with anything, I'm a big supporter of this API.

**Comment by itsuwari** [comment]
POE.com might be a nice addition, since OpenAI and Google Claude are both alive there.

**Comment by noseon** [comment]
It wouldn't be an idea to look like the proxy on the cloudflare Workers system if the code is simple everyone can create yours to use

**Comment by acheong08** [comment]
I mean the bypass method is patched.

**Comment by acheong08** [comment]
> POE.com might be a nice addition, since OpenAI and Google Claude are both alive there.

POE/POE.py at main · vaibhavk97/Poe

**Comment by 7trail** [comment]
Now's a good time to mention that POE.com actually has ChatGPT built into it. It also does not have a visible rate limit. Adding access to POE through here would be nice.

**Comment by acheong08** [comment]
> Now's a good time to mention that POE.com actually has ChatGPT built into it. It also does not have a visible rate limit. Adding access to POE through here would be nice.

Will work on it.

**Comment by noseon** [comment]
https://github.com/mmabrouk/chatgpt-wrapper/blob/main/chatgpt_wrapper/gpt_api.py would it be a good idea to put the vercel.app proxy on this site?

**Comment by ouwx** [comment]
acheong08，wait you fix

**Comment by acheong08** [comment]
Unlikely for this to get fixed. V2 and V3 are currently the only methods

**Comment by acheong08** [comment]
The API key for V2 should be `pk-TNkDYMHpJKKuSfiAWJlUCKQSnKluoZxvLGKRPnPzxCDPdVxs`

**Comment by NamiKaze7** [comment]
Build an available endpoint maybe ?

**Comment by seyrenus** [comment]
Is there any other alternative method to get original web chatgpt answer other than open chat.openai.com? the gpt-3.5-turbo api is still different from web chatgpt.

**Comment by Liu-Angelo** [comment]
> V2 的 API 密钥应该是`pk-TNkDYMHpJKKuSfiAWJlUCKQSnKluoZxvLGKRPnPzxCDPdVxs`

python3 -m revChatGPT.V2 --api_key=pk-TNkDYMHpJKKuSfiAWJlUCKQSnKluoZxvLGKRPnPzxCDPdVxs 

        ChatGPT - A command-line interface to OpenAI's ChatGPT (https://chat.openai.com/chat)
        Repo: github.com/acheong08/ChatGPT
        
Logging in...
Logged in

Type '!help' to show a full list of commands
Press enter twice to submit your question.


You:
写一个二分查找
 
 
revChatGPT==3.1.8
python3 -m revChatGPT.V2 也不行哈

**Comment by acheong08** [comment]
It's working. I need to update the instructions. `alt+enter` to submit

**Comment by renewboy** [comment]
Is this endpoint still unavailable? Any plan to fix it?

**Comment by noseon** [comment]
in my view the owner should show how to make the proxy for anyone who wants to use it somewhere. because there are a lot of people using his proxy and in this case there is a cost to maintain the proxy.

**Comment by acheong08** [comment]
> in my view the owner should show how to make the proxy for anyone who wants to use it somewhere. because there are a lot of people using his proxy and in this case there is a cost to maintain the proxy.

We did that with V1. Look how fast it got patched. (A few hours)

**Comment by acheong08** [comment]
It seems everything is overloaded though. Waiting for @PawanOsman to either scale up or share source code with people who can

**Comment by acheong08** [comment]
Oh shit someone from the community found a new bypass. Will write up a server for that soon

**Reply by itsuwari** [reply]
Just wondering what this proxy is for, I tried to use cf-scrape instead of original requests and it worked perfectly. I can cooperate with you to make a binary version of the proxy/setup proxy server, but there is still possibility someone from openai/CF may construct requests to find out what’s happening

**Reply by acheong08** [reply]
> Just wondering what this proxy is for

To avoid OpenAI from patching

>  I can cooperate with you to make a binary version of the proxy/setup proxy server

It is a binary by default since the server is written in Go

>  but there is still possibility someone from openai/CF may construct requests to find out what’s happening

Yup. There's also mitmproxy

> I tried to use cf-scrape instead of original requests and it worked perfectly. 

👀 that works? damn that would make things so much easier. Will try it out.

**Reply by acheong08** [reply]
- https://github.com/Anorov/cloudflare-scrape/issues/347

**Reply by acheong08** [reply]
Can you tell me what version of cf-scrape works?

**Reply by itsuwari** [reply]
I am currently using the branch https://github.com/Anorov/cloudflare-scrape/pull/373.  As I am not sending a large volume of requests, I presume that is why I am not encountering hcaptcha.I am currently exploring the Captcha Solver plugin from 2captcha-com within selenium-requests, might be a more effective approach, and not expensive.

**Reply by itsuwari** [reply]
I replaced standard chrome drive with https://github.com/ultrafunkamsterdam/undetected-chromedriver, made bypassing easier, currently works with my home broadband IP, but greeted with captchas on M247 VPS (not blocked by OpenAI).

**Reply by itsuwari** [reply]
I also tried https://github.com/Xewdy444/CF-Clearance-Scraper(Firefox) to get cf_clearance, then proceeded with auth, seems working on my home broadband too.

**Reply by acheong08** [reply]
I'll try those once my current bypass gets patched

**Reply by itsuwari** [reply]
Hope the bypass won't be patched. I tried another Cloudflare solver(https://github.com/VeNoMouS/cloudscraper), this one doesn't even require a browser, and support for captcha solver plugins is seamless. Here is a working fork https://github.com/freeotaku/ChatGPT-CFbypass

**Reply by s524797336** [reply]
@itsuwari The code is not working.
AttributeError: 'CloudScraper' object has no attribute 'Session'

**Reply by itsuwari** [reply]
Sorry for the stupid mistake, I was using a modified version. Fixed and tested.

**Reply by itsuwari** [reply]
Just a reminder: the auth library imported in V1 should be modified accordingly too. I have captcha solver credentials embed into my code, once I get everything cleaned up I'll share the code.

**Reply by acheong08** [reply]
I published the proxy code https://github.com/acheong08/ChatGPT-Proxy-V4

**Reply by acheong08** [reply]
I found new bypass so it's ok if this gets patched

**Reply by acheong08** [reply]
I'm more testing to see how OpenAI would respond

**Reply by itsuwari** [reply]
This is so much lighter than using selenium, and works perfectly with an IPv4 which always gets captchas, I have deployed the proxy yesterday, after >1k requests(5 accounts, paid, 8h) then prompted with CF1020(sharing a /128 v6 address), then I allocated different /128 to each proxy and its running great without any disruption so far(>400reqs 12h).

**Comment by s524797336** [comment]
@acheong08 Can you provide a proxy bin just like before?

**Reply by acheong08** [reply]
No. Binary got reverse engineered

**Comment by noseon** [comment]
makes a pyc in python with very friendly configuration. because it can run both locally and on a site that runs python and becomes a proxy. so you get rid of having to pay a proxy or you try to make it run on cloudflare on Workers ai each person does their own.

**Comment by acheong08** [comment]
> makes a pyc in python with very friendly configuration. because it can run both locally and on a site that runs python and becomes a proxy. so you get rid of having to pay a proxy or you try to make it run on cloudflare on Workers ai each person does their own.

Are you willing to cooperate with me? I don't have the time to do too much

**Reply by ernongxizhu** [reply]
I can spare some time to do it

**Reply by noseon** [reply]
> > makes a pyc in python with very friendly configuration. because it can run both locally and on a site that runs python and becomes a proxy. so you get rid of having to pay a proxy or you try to make it run on cloudflare on Workers ai each person does their own.
> 
> Are you willing to cooperate with me? I don't have the time to do too much

I will only have time in 10 days so I can help.

**Reply by acheong08** [reply]
Check out TLS spoofing on NodeJS. If you guys figure that out, it will be easy to port

**Comment by acheong08** [comment]
The code is written in Go btw. Not sure how to port it to JS because it involves TLS spoofing

**Reply by noseon** [reply]
> The code is written in Go btw. Not sure how to port it to JS because it involves TLS spoofing

https://blog.cloudflare.com/node-js-support-cloudflare-workers/  nodejs if you manage to use it on the site it will be good.

**Reply by acheong08** [reply]
NodeJS does not support streaming while spoofing TLS that I know of. If you can figure that out, it would be nice
